app/services/attribute_asserter.rb

```diff
@@ -0,0 +1,66 @@
+class AttributeAsserter
+  attr_accessor :user, :service_provider, :authn_request
+
+  def initialize(user, service_provider, authn_request)
+    self.user = user
+    self.service_provider = service_provider
+    self.authn_request = authn_request
+  end
+
+  def build
+    attrs = default_attrs
+    add_email(attrs) if bundle.include? 'email'
+    add_mobile(attrs) if bundle.include? 'mobile'
+    add_bundle(attrs) if user.verified?
+    user.asserted_attributes = attrs
+  end
+
+  private
+
+  def default_attrs
+    {
+      uuid: {
+        getter: :uuid,
+        name_format: Saml::XML::Namespaces::Formats::NameId::PERSISTENT,
+        name_id_format: Saml::XML::Namespaces::Formats::NameId::PERSISTENT
+      }
+    }
+  end
+
+  def add_bundle(attrs)
+    bundle.each do |attr|
+      attrs[attr] = { getter: deferred_profile_attribute(attr) }
+    end
+  end
+
+  def deferred_profile_attribute(attr)
+    -> (principal) { principal.active_profile[attr] }
```

Comment:
Oh.. exciting new syntax.  Not seen that before.  The name of this method didn't initial convey to me what it does.  What about "attribute_getter_function"  or "build_attribute_getter"?  Or perhaps inline the dash rocket syntax?